federatedscope/cl/fedgc/worker.py

```diff
@@ -0,0 +1,226 @@
+                        f'#{self.state}) -------------')
+                    # Clean the msg_buffer
+                    self.msg_buffer['train'][self.state - 1].clear()
+                    self.msg_buffer['train'][self.state] = dict()
+                    self.staled_msg_buffer.clear()
+                    # Start a new training round
+                    
+
+                else:
+                    # Final Evaluate
+                    logger.info('Server: Training is finished! Starting '
+                                'evaluation.')
+                    self.eval()
+
+            else:  # in the evaluation process
+                # Get all the message & aggregate
+                formatted_eval_res = self.merge_eval_results_from_all_clients()
+                self.history_results = merge_dict(self.history_results,
+                                                  formatted_eval_res)
+                self.check_and_save()
+                
+    def callback_funcs_model_para(self, message: Message):
+        """
+        The handling function for receiving model parameters, which triggers
+            check_and_move_on (perform aggregation when enough feedback has
+            been received).
+        This handling function is widely used in various FL courses.
+
+        Arguments:
+            message: The received message, which includes sender, receiver,
+                state, and content. More detail can be found in
+                federatedscope.core.message
+        """
+        if self.is_finish:
+            return 'finish'
+
+        round = message.state
+        sender = message.sender
+        timestamp = message.timestamp
+        content = message.content
+        self.sampler.change_state(sender, 'idle')
+
+        # update the currency timestamp according to the received message
+        assert timestamp >= self.cur_timestamp  # for test
+        self.cur_timestamp = timestamp
+
+        if round == self.state:
+            if round not in self.msg_buffer['train']:
+                self.msg_buffer['train'][round] = dict()
+            # Save the messages in this round
+            self.msg_buffer['train'][round][sender] = content
+        elif round >= self.state - self.staleness_toleration:
+            # Save the staled messages
+            self.staled_msg_buffer.append((round, sender, content))
+        else:
+            # Drop the out-of-date messages
+            logger.info(f'Drop a out-of-date message from round #{round}')
+            self.dropout_num += 1
+
+        if self._cfg.federate.online_aggr:
+            self.aggregator.inc(content[:2])
+
+        move_on_flag = self.check_and_move_on()
+        if self._cfg.asyn.use and self._cfg.asyn.broadcast_manner == \
+                'after_receiving':
+            self.broadcast_model_para(msg_type='model_para',
+                                      sample_client_num=1)
+
+        return move_on_flag
+
+
+class GlobalContrastFLClient(Client):
```

Comment:
put these two classes into server.py and client.py

Reply:
finish